jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -43,58 +46,77 @@ public class JmxScraper {
+      AutoConfiguredOpenTelemetrySdk.builder()
+          .addPropertiesSupplier(new PropertiesSupplier(argsConfig))
+          .addPropertiesCustomizer(configCustomizer)
+          .setResultAsGlobal()
+          .build();
+
+      JmxScraperConfig scraperConfig = configCustomizer.getScraperConfig();
+
+      long exportSeconds = scraperConfig.getSamplingInterval().toMillis() / 1000;
+      logger.log(Level.INFO, "metrics export interval (seconds) =  " + exportSeconds);
 
       JmxMetricInsight service =
           JmxMetricInsight.createService(
-              GlobalOpenTelemetry.get(), config.getIntervalMilliseconds());
-      JmxConnectorBuilder connectorBuilder = JmxConnectorBuilder.createNew(config.getServiceUrl());
+              GlobalOpenTelemetry.get(), scraperConfig.getSamplingInterval().toMillis());
+      JmxConnectorBuilder connectorBuilder =
+          JmxConnectorBuilder.createNew(scraperConfig.getServiceUrl());
 
-      Optional.ofNullable(config.getUsername()).ifPresent(connectorBuilder::withUser);
-      Optional.ofNullable(config.getPassword()).ifPresent(connectorBuilder::withPassword);
+      Optional.ofNullable(scraperConfig.getUsername()).ifPresent(connectorBuilder::withUser);
+      Optional.ofNullable(scraperConfig.getPassword()).ifPresent(connectorBuilder::withPassword);
 
-      JmxScraper jmxScraper = new JmxScraper(connectorBuilder, service, config);
+      JmxScraper jmxScraper = new JmxScraper(connectorBuilder, service, scraperConfig);
       jmxScraper.start();
-
+    } catch (ConfigurationException e) {
+      logger.log(Level.SEVERE, "ERROR: invalid configuration ", e);
+      System.exit(1);
     } catch (ArgumentsParsingException e) {
-      System.err.println("ERROR: " + e.getMessage());
-      System.err.println(
+      logger.log(Level.SEVERE, "ERROR: invalid configuration provided through arguments", e);
+      logger.info(
           "Usage: java -jar <path_to_jmxscraper.jar> "
               + "-config <path_to_config.properties or - for stdin>");
       System.exit(1);
-    } catch (ConfigurationException e) {
-      System.err.println(e.getMessage());
-      System.exit(1);
     } catch (IOException e) {
-      System.err.println("Unable to connect " + e.getMessage());
+      logger.log(Level.SEVERE, "Unable to connect ", e);
       System.exit(2);
     } catch (RuntimeException e) {
-      e.printStackTrace(System.err);
+      logger.log(Level.SEVERE, e.getMessage(), e);
       System.exit(3);
     }
   }
 
+  // package private for testing
+  static void propagateToSystemProperties(Properties properties) {
+    for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+      String key = entry.getKey().toString();
+      String value = entry.getValue().toString();
+      if (key.startsWith("javax.net.ssl.keyStore") || key.startsWith("javax.net.ssl.trustStore")) {
+        if (System.getProperty(key) == null) {
+          System.setProperty(key, value);
+        }
+      }
+    }
+  }
```

Comment:
[for reviewer] propagation of system properties is only used to set the java keystore/truststore options from the program arguments/standard input to the global JVM settings, so it was simpler to move it here. In practice this will likely be tested when we add tests with custom keystore/truststore by setting those configuration options from the standard input or the properties file.